chain-impl-mockchain/src/leadership/bft.rs

```diff
@@ -14,6 +14,12 @@ pub type BftVerificationAlg = Ed25519;
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
 pub struct LeaderId(pub(crate) PublicKey<BftVerificationAlg>);
 
+impl From<[u8; 32]> for LeaderId {
+    fn from(v: [u8; 32]) -> LeaderId {
+        LeaderId(PublicKey::from_binary(&v[..]).expect("leader-id invalid format"))
+    }
+}
```

Comment:
what about `TryFrom` instead of `From` to avoid a partial function here?

Reply:
the conversion cannot fail. this is what I've been saying that I need new interfaces in chain-crypto to do these things